libs/langchain-azure-cosmosdb/src/caches.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  BaseCache,
+  deserializeStoredGeneration,
+  getCacheKey,
+  serializeGeneration,
+} from "@langchain/core/caches";
+import { Generation } from "@langchain/core/outputs";
+import { Document } from "@langchain/core/documents";
+import { EmbeddingsInterface } from "@langchain/core/embeddings";
+import { CosmosClient, CosmosClientOptions } from "@azure/cosmos";
+import { DefaultAzureCredential } from "@azure/identity";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import {
+  AzureCosmosDBNoSQLConfig,
+  AzureCosmosDBNoSQLVectorStore,
+} from "./azure_cosmosdb_nosql.js";
+
+const USER_AGENT_SUFFIX = "LangChain-CDBNoSQL-SemanticCache-JavaScript";
```

Comment:
If possible, it would be preferable to use `langchainjs-azure-cosmosdb-nosql-semanticcache` (see previous comment)

Reply:
@sinedied we are following a user agent pattern across all ai integrations, and we came up with "LangChain-CDBNoSQL-VectorStore-JavaScript"